webapp-django/crashstats/home/static/home/js/home.js

```diff
@@ -0,0 +1,386 @@
+/*global MG: true, socorro: true */
+
+$(function () {
+    'use strict';
+
+    // A polyfill for `endsWith`
+    if (!String.prototype.endsWith) {
+        String.prototype.endsWith = function(searchString, position) {
+            var subjectString = this.toString();
+            if (typeof position !== 'number' || !isFinite(position) || Math.floor(position) !== position || position > subjectString.length) {
+                position = subjectString.length;
+            }
+            position -= searchString.length;
+            var lastIndex = subjectString.indexOf(searchString, position);
+            return lastIndex !== -1 && lastIndex === position;
+        };
+    }
+
+    var COLORS = ['#6a3d9a', '#e31a1c', '#008800', '#1f78b4'];
+
+    var container = $('#homepage-graph-container');
+    var superSearchApiUrl = container.data('supersearch-api');
+    var productBuildTypesApiUrl = container.data('product-build-types-api');
+    var adiApiUrl = container.data('adi-api');
+
+    var product = container.data('product');
+    var versions = container.data('versions');
+    var platforms = container.data('platforms');
+    var duration = container.data('duration');
+
+    var pageTitle = $('title').text();
+
+    // Versions ending in -b are "magic" versions that will be split into all
+    // of the beta versions starting with that same version number.
+    var betaVersions = [];
+    for (var i = versions.length - 1; i >= 0; i--) {
+        if (versions[i].endsWith('b')) {
+            betaVersions.push(versions[i]);
+        }
+    }
+
+    // Set default dates.
+    var endDate = new Date();
+    endDate.setDate(endDate.getDate() - 1);  // We start with yesterday.
+    var startDate = new Date();
+    startDate.setDate(endDate.getDate() - duration);
+
+    var endDateStr = dateIsoFormat(endDate);
+    var startDateStr = dateIsoFormat(startDate);
+
+    /**
+     * Data received from services, after being transformed, will be stored
+     * here. Once we have received everything, we'll be able to combine all
+     * those data into the final data we want to show as a graph.
+     */
+    var dataStore = {
+        superSearch: null,
+        adi: null,
+        builds: null,
+    };
+
+    /**
+     * Return a string with only the date part of a Date object, in ISO format.
+     */
+    function dateIsoFormat(date) {
+        return date.toISOString().substring(0, 10);
+    }
+
+    /**
+     * History change handler. Reloads the graph's data.
+     */
+    window.onpopstate = function (e) {
+        var days = e.state.duration;
+
+        $('#duration .selected').removeClass('selected');
+        $('#duration .days-' + days).addClass('selected');
+
+        updateDates(days);
+        loadData();
+    };
+
+    /**
+     * Click event handler for the duration menu. Changes the history and
+     * reloads the graph's data.
+     */
+    $('#duration a').click(function (e) {
```

Comment:
Change this to
```javascript
$('#duration').on('click', 'a', function(e) {
```
That's more performant. It's better to have 1 event handler listening and within it, delegate based on what was clicked.